### PR TITLE
Added Result extension

### PR DESCRIPTION
**Pull request by ShihabMehboob**
In relation to the following issue: https://github.com/IBM-Swift/SwiftyRequest/issues/27#event-1664040225

### CODE REVIEW

File: Sources/SwiftyRequest/ResultExtension.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright IBM Corporation 2017
```
Comment (ianpartridge) [inline Sources/SwiftyRequest/ResultExtension.swift]: 2018

Comment (ShihabMehboob) [reply Sources/SwiftyRequest/ResultExtension.swift]: Changed this now.

**Comment by ianpartridge** [comment]
@ShihabMehboob I didn't realise this hadn't been merged, and now it's failing to build. Can you take a look please?

**Comment by ianpartridge** [comment]
Thanks, but I'm confused...! Why did the build on 01165a1 pass but the next one fail? Are we testing against a different Swift version?

**Comment by ShihabMehboob** [comment]
@ianpartridge The only difference seems to have been that the repo got updated to Swift 4.1.2 from 4.0.3, which seems to have caused this requirement for public modifiers not being allowed to be used in protocols and extensions that declare protocol conformances.

**Comment by ianpartridge** [comment]
Oh OK thanks! Do we not test on Swift 4.0.3 any more then? Seems like we should?

**Comment by ShihabMehboob** [comment]
@ianpartridge They seemed to have been removed with the 4.2 changes (when most repos were updated to it recently), added it back now.

**Pull request merged**